presto-main/src/main/java/io/prestosql/server/protocol/QueryResultRows.java

```diff
@@ -107,7 +107,14 @@ public long getTotalRowsCount()
     public Optional<Long> getUpdateCount()
     {
         // We should have exactly single bigint value as an update count.
-        if (totalRows != 1 || columns.size() != 1 || !columns.get(0).getType().equals(StandardTypes.BIGINT)) {
+        if (totalRows != 1 || columns.isEmpty()) {
+            return Optional.empty();
+        }
+
+        verify(columns.isPresent(), "columns are not present");
+        List<Column> currentColumns = columns.get();
```

Comment:
`current` implies this changes over time, while column information does not.

you can use `columns` for the local var (yes, shadowing the field)